packages/hash/api/src/model/entity.model.ts

```diff
@@ -535,11 +536,23 @@ class __Entity {
           const systemAccountId = await client.getSystemAccountId();
 
           const name = capitalizeComponentName(componentId);
+
+          // ensure a trailing a trailing slash on componentId
+          const componentIdUrl = new URL(
+            "./block-schema.json",
+            `${componentId.replace(/\/+$/, "")}/`,
+          );
+
+          const blockSchema = await (await fetch(componentIdUrl.href))
+            .json()
+            .catch((_) => ({}));
+
+          // Creation of an EntityType validates schema.
           entityTypeWithComponentId = await EntityType.create(client, {
             accountId: systemAccountId,
             createdByAccountId: params.user.accountId,
             name,
-            schema: { componentId },
+            schema: { ...blockSchema, componentId },
```

Comment:
Is it `blockSchema.properties` we want to merge here? I'm unsure?

Reply:
I believe it would be the whole schema, as we the JSONSchema library has been extended to support `componentId` at the top level along with all the other fields.